liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/CreateServerMojo.java

```diff
@@ -174,4 +175,39 @@ private File getFileFromConfigDirectory(String file, File def) {
         } 
         return null;
     }
+    
+    /*
+     * return the filename of the project artifact to be installed by install-apps goal
+     */
+    private String getApplicationFilename() {
+        // project artifact has not be created yet when create-server goal is called in pre-package phase
+        String name = project.getBuild().getFinalName();
+        if (stripVersion) {
+            int versionBeginIndex = project.getBuild().getFinalName().lastIndexOf("-" + project.getVersion());
+            if ( versionBeginIndex != -1) {
+                name = project.getBuild().getFinalName().substring(0, versionBeginIndex);
+            }
+        }
+        
```

Comment:
Yes, I did notice that.  The installAppMojo makes no assumption what the file type is.  At the point in time the .war extension is already on the artifact name,  and I just reserved the extension that was already there.